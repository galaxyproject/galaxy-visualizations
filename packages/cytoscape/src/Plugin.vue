<script setup lang="ts">
import Cytoscape, { type Core, type ElementDefinition, type ElementsDefinition } from "cytoscape";
import { onMounted, onUnmounted, ref, watch } from "vue";

import { parseSIF, runSearchAlgorithm, runTraversalType, styleGenerator } from "./utils";

const TEST_URL = "http://cdn.jsdelivr.net/gh/galaxyproject/galaxy-test-data/1.cytoscapejson";

type Settings = {
    directed: boolean;
    layout_name: "breadthfirst" | "circle" | "concentric" | "cose" | "grid" | "null" | "preset" | "random";
    curve_style: "haystack" | "straight" | "bezier" | "unbundled-bezier" | "segments" | "taxi";
    color_picker_nodes: string;
    color_picker_edges: string;
    color_picker_highlighted: string;
    graph_traversal?: "successors" | "predecessors" | "outgoers" | "incomers" | "roots" | "leaves";
    search_algorithm?: "bfs" | "dfs" | "kruskal" | "astar";
};

interface Props {
    root: string;
    datasetId: string;
    datasetUrl: string;
    settings: Settings;
}

const props = defineProps<Props>();

const viewport = ref(null);

const cytoscape = ref<Core | null>(null);
const dataset = ref<ElementsDefinition | ElementDefinition[]>();

async function render() {
    cytoscape.value = Cytoscape({
        container: viewport.value,
        elements: dataset.value,
        layout: {
            name: props.settings.layout_name,
            idealEdgeLength: () => 100,
            nodeOverlap: 20,
        },
        minZoom: 0.1,
        maxZoom: 50,
        style: styleGenerator(props.settings),
    });

    if (props.settings.search_algorithm === "kruskal") {
        //@ts-ignore
        cytoscape.value.elements().kruskal().edges().addClass("searchpath");
    }

    cytoscape.value.on("tap", "node", (e) => {
        const ele = e.target;

        const search_algorithm = props.settings.search_algorithm;
        const traversal_type = props.settings.graph_traversal;

        if (cytoscape.value && search_algorithm) {
            runSearchAlgorithm(cytoscape.value, ele.id(), search_algorithm);
        } else if (cytoscape.value && traversal_type) {
            runTraversalType(cytoscape.value, ele.id(), traversal_type);
        }
    });
}

async function getDataset() {
    try {
        const datasetUrl = props.datasetId === "__test__" ? TEST_URL : props.datasetUrl;
        const response = await fetch(datasetUrl);
        if (response.ok) {
            const content = await response.json();
            if (content.file_ext === "sif") {
                dataset.value = parseSIF(content).content;
            } else {
                dataset.value = content.elements ?? content;
            }
        } else {
            console.error("Fetch failed", response);
        }
    } catch (err) {
        console.error("Dataset fetch error", err);
    }
}

watch(
    () => props,
    () => render(),
    { deep: true },
);

onMounted(async () => {
    await getDataset();
    await render();
    window.addEventListener("resize", function () {
        cytoscape.value?.resize();
    });
});

onUnmounted(() => {
    window.removeEventListener("resize", function () {
        console.log("Removing resize event listener");
    });

    cytoscape.value?.destroy();
});
</script>

<template>
    <div class="h-screen overflow-hidden" ref="viewport"></div>
</template>
