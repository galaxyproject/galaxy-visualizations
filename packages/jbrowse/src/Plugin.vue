<script setup>
import { onMounted, ref, watch } from "vue";
import { renderJBrowse } from "./render";

const props = defineProps({
    datasetId: String,
    datasetUrl: String,
    root: String,
    settings: Object,
    specs: Object,
    tracks: Array,
});

const viewport = ref(null);

function render() {
    const config = JSON.parse(JSON.stringify(props.settings.jbrowseConfig));
    if (viewport.value) {
        renderJBrowse(viewport.value, config);
    }
}

onMounted(() => {
    render();
});

watch(
    () => props,
    () => render(),
    { deep: true },
);
</script>

<template>
    <div ref="viewport"></div>
</template>
