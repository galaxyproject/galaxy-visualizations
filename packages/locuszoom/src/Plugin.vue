<script setup>
import { onMounted, watch, ref } from "vue";
import "locuszoom/dist/locuszoom.css";
import LocusZoom from "locuszoom/esm";
import LzTabixSource from "locuszoom/esm/ext/lz-tabix-source";
import { makeGWASParser } from "locuszoom/esm/ext/lz-parsers";
import { plotUpdatesUrl, plotWatchesUrl } from "locuszoom/esm/ext/lz-dynamic-urls";

LocusZoom.use(LzTabixSource);

window.nowrap = false;
window.mode = false;

const COLOR = "#E30A17";
const MAX_RANGE = 10000000;
const SIZE = 40;

const props = defineProps({
    datasetId: String,
    datasetUrl: String,
    root: String,
    settings: Object,
    specs: Object,
    tracks: Array,
});

const errorMessage = ref("");

function render() {
    const id = props.settings.tabix?.id;
    const chrIn = props.settings.chromosome;
    const startIn = props.settings.start;
    const endIn = props.settings.end;
    const chrom_col = props.settings.chrom_col;
    const pos_col = props.settings.pos_col;
    const ref_col = props.settings.ref_col;
    const alt_col = props.settings.alt_col;
    const pvalue_col = props.settings.pvalue_col;
    const is_neg_log_pvalue = props.settings.is_neg_log_pvalue;
    const beta_col = props.settings.beta_col;
    const stderr_beta_col = props.settings.stderr_beta_col;
    if (!id) {
        errorMessage.value = "Please select a Tabix file.";
        return;
    }
    if (endIn - startIn > MAX_RANGE) {
        errorMessage.value = "We cannot output more than 10Mb at a time!";
        return;
    }
    if (endIn <= startIn) {
        errorMessage.value = "Invalid input: End position must be bigger than start position!";
        return;
    }
    errorMessage.value = "";
    const gwasParser = makeGWASParser({
        chrom_col,
        pos_col,
        ref_col,
        alt_col,
        pvalue_col,
        is_neg_log_pvalue,
        beta_col,
        stderr_beta_col,
    });
    let data_sources = new LocusZoom.DataSources().add("assoc", [
        "TabixUrlSource",
        {
            url_data: `${props.root}api/datasets/${props.datasetId}/display`,
            url_tbi: `${props.root}api/datasets/${id}/display`,
            parser_func: gwasParser,
            overfetch: 0,
        },
    ]);
    let stateUrlMapping = { chr: "chrom", start: "start", end: "end", ldrefvar: "ld_variant" };
    let association_data_layer_mods = {
        id: "associationpvalues_",
        name: "association",
        point_shape: "circle",
        point_size: SIZE,
        color: COLOR,
        legend: [{ shape: "circle", color: COLOR, size: SIZE, label: "phen", class: "lz-data_layer-scatter" }],
        tooltip: {
            closable: true,
            show: { or: ["highlighted", "selected"] },
            hide: { and: ["unhighlighted", "unselected"] },
            html: `<strong>phen</strong><br><strong>{{assoc:variant|htmlescape}}</strong><br>
                P Value: <strong>{{assoc:log_pvalue|logtoscinotation|htmlescape}}</strong><br>
                Ref. Allele: <strong>{{assoc:ref_allele|htmlescape}}</strong><br>`,
        },
    };
    const data_layer = {
        ...LocusZoom.Layouts.get("data_layer", "association_pvalues", association_data_layer_mods),
        namespace: { assoc: "assoc" },
        data_operations: [],
    };
    const panel = LocusZoom.Layouts.get("panel", "association", {
        data_layers: [data_layer],
    });
    let plot = LocusZoom.populate("#lz-plot", data_sources, {
        state: { chr: chrIn, start: startIn, end: endIn },
        responsive_resize: true,
        panels: [panel],
        toolbar: LocusZoom.Layouts.get("toolbar", "standard_plot"),
    });
    plotUpdatesUrl(plot, stateUrlMapping);
    plotWatchesUrl(plot, stateUrlMapping);

    // manually set size of svg container
    document.querySelector("#lz-plot_svg").style.height = "calc(100vh - 50px)";
}

onMounted(() => {
    render();
});

watch(
    () => props,
    () => {
        render();
    },
    { deep: true },
);
</script>

<template>
    <div id="lz-plot" class="h-100"></div>
    <div v-if="errorMessage" class="mt-4 p-3 border bg-red-100 text-red-700 text-sm text-center rounded">
        {{ errorMessage }}
    </div>
</template>

<style>
.lz-panel-toolbar {
    display: none;
}
</style>
