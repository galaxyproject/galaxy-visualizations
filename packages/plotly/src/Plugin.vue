<script setup>
import Plotly from "plotly.js-dist";
import { onMounted, ref, watch } from "vue";

import plotlySurface from "@/variants/surface";
import plotlyBasics from "@/variants/basics";
import plotlyBox from "@/variants/box";
import plotlyHeatmap from "@/variants/heatmap";
import plotlyHistogram from "@/variants/histogram";

const props = defineProps({
    datasetId: String,
    datasetUrl: String,
    root: String,
    settings: Object,
    specs: Object,
    tracks: Array,
});

const message = ref();
const viewport = ref(null);

async function render() {
    let wrapper = null;
    switch (props.specs?.variant) {
        case "box":
            wrapper = plotlyBox;
            break;
        case "histogram":
            wrapper = plotlyHistogram;
            break;
        case "heatmap":
            wrapper = plotlyHeatmap;
            break;
        case "surface":
            wrapper = plotlySurface;
            break;
        default:
            wrapper = plotlyBasics;
    }
    try {
        message.value = "";
        const { data, layout, config } = await wrapper(props.datasetId, props.settings, props.tracks);
        Plotly.newPlot(viewport.value, data, layout, config);
    } catch (e) {
        message.value = `Failed to render: ${e}`;
        console.error(e);
    }
}

onMounted(() => {
    render();
});

watch(
    () => props,
    () => render(),
    { deep: true },
);
</script>

<template>
    <div v-if="message" class="bg-sky-100 border border-sky-200 p-1 rounded text-sky-800 text-sm">
        {{ message }}
    </div>
    <div ref="viewport" class="h-screen" />
</template>
