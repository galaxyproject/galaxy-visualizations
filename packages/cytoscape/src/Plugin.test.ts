import axios from "axios";
import { mount } from "@vue/test-utils";
import { describe, expect, it, vi } from "vitest";

import Plugin from "./Plugin.vue";

const mockGetResponse = {
    data: {
        format_version: "1.0",
        generated_by: "cytoscape-3.1.1",
        elements: {
            nodes: [{ data: { id: "n0", name: "node0" } }, { data: { id: "n1", name: "node1" } }],
            edges: [{ data: { source: "n0", target: "n1" } }],
        },
        data: {
            name: "mock",
        },
    },
};

vi.spyOn(axios, "get").mockResolvedValue(mockGetResponse);

describe("Plugin", () => {
    it("should render correctly", () => {
        HTMLCanvasElement.prototype.getContext = vi.fn(() => {
            return {
                fillRect: vi.fn(),
                clearRect: vi.fn(),
                getImageData: vi.fn(() => {
                    return {
                        data: new Array(100).fill(0),
                    };
                }),
                putImageData: vi.fn(),
                createImageData: vi.fn(() => {
                    return {
                        data: new Array(100).fill(0),
                    };
                }),
                setTransform: vi.fn(),
                drawImage: vi.fn(),
                save: vi.fn(),
                context: {
                    translate: vi.fn(),
                },
                fillText: vi.fn(),
                restore: vi.fn(),
                measureText: vi.fn(() => {
                    return { width: 0 };
                }),
            };
        });

        const wrapper = mount(Plugin, {
            props: {
                root: "/",
                tracks: [],
                credentials: "include",
                datasetId: "MY_DATASET_ID",
                datasetUrl: "network.json",
                settings: {
                    curve_style: "haystack",
                    layout_name: "preset",
                    directed: false,
                    search_algorithm: undefined,
                    graph_traversal: undefined,
                    color_picker_nodes: "#548DB8",
                    color_picker_edges: "#A5A5A5",
                    color_picker_highlighted: "#C00000",
                },
            },
        });

        expect(wrapper).toMatchSnapshot();
    });
});
