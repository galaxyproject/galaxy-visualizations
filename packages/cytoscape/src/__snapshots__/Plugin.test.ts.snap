// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Plugin > should render correctly 1`] = `
VueWrapper {
  "__app": {
    "_component": {
      "name": "VTU_ROOT",
      "render": [Function],
      "setup": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        class="h-screen overflow-hidden"
        credentials="include"
      />
    </div>,
    "_context": {
      "app": [Circular],
      "components": {
        "transition": {
          "name": "transition",
        },
        "transition-group": {
          "name": "transition-group",
        },
      },
      "config": {
        "compilerOptions": {},
        "errorHandler": undefined,
        "globalProperties": {},
        "isNativeTag": [Function],
        "optionMergeStrategies": {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": {},
      "emitsCache": WeakMap {},
      "mixins": [
        {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": {},
      "reload": [Function],
    },
    "_instance": {
      "a": null,
      "accessCache": {},
      "appContext": {
        "app": [Circular],
        "components": {
          "transition": {
            "name": "transition",
          },
          "transition-group": {
            "name": "transition-group",
          },
        },
        "config": {
          "compilerOptions": {},
          "errorHandler": undefined,
          "globalProperties": {},
          "isNativeTag": [Function],
          "optionMergeStrategies": {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": {},
        "emitsCache": WeakMap {},
        "mixins": [
          {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": {
        "VTU_COMPONENT": {},
      },
      "da": null,
      "data": {},
      "devtoolsRawSetupState": {
        "VTU_COMPONENT": RefImpl {
          "__v_isRef": true,
          "__v_isShallow": false,
          "_rawValue": {},
          "_value": {},
          "dep": undefined,
        },
      },
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "_depsLength": 9,
        "_dirtyLevel": 0,
        "_runnings": 0,
        "_shouldSchedule": false,
        "_trackId": 1,
        "active": true,
        "allowRecurse": true,
        "deps": [
          Map {
            [Circular] => 1,
          },
          Map {
            [Circular] => 1,
          },
          Map {
            [Circular] => 1,
          },
          Map {
            [Circular] => 1,
          },
          Map {
            [Circular] => 1,
          },
          Map {
            [Circular] => 1,
          },
          Map {
            [Circular] => 1,
          },
          Map {
            [Circular] => 1,
          },
          Map {
            [Circular] => 1,
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "scheduler": [Function],
        "trigger": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": {},
      "propsDefaults": {},
      "propsOptions": [
        {},
        [],
      ],
      "provides": {},
      "proxy": {
        "VTU_COMPONENT": {},
      },
      "refs": {
        "VTU_COMPONENT": {},
      },
      "render": [Function],
      "renderCache": [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "_active": true,
        "cleanups": [],
        "detached": true,
        "effects": [
          ReactiveEffect {
            "_depsLength": 9,
            "_dirtyLevel": 0,
            "_runnings": 0,
            "_shouldSchedule": false,
            "_trackId": 1,
            "active": true,
            "allowRecurse": true,
            "deps": [
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "scheduler": [Function],
            "trigger": [Function],
          },
        ],
        "parent": undefined,
      },
      "setupContext": null,
      "setupState": {
        "VTU_COMPONENT": {},
      },
      "slots": {},
      "sp": null,
      "subTree": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": {
          "a": null,
          "accessCache": {},
          "appContext": {
            "app": [Circular],
            "components": {
              "transition": {
                "name": "transition",
              },
              "transition-group": {
                "name": "transition-group",
              },
            },
            "config": {
              "compilerOptions": {},
              "errorHandler": undefined,
              "globalProperties": {},
              "isNativeTag": [Function],
              "optionMergeStrategies": {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": {},
            "emitsCache": WeakMap {},
            "mixins": [
              {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": {
            "credentials": "include",
          },
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": {},
          "ctx": {
            "datasetId": "MY_DATASET_ID",
            "datasetUrl": "network.json",
            "root": "/",
            "settings": {
              "color_picker_edges": "#A5A5A5",
              "color_picker_highlighted": "#C00000",
              "color_picker_nodes": "#548DB8",
              "curve_style": "haystack",
              "directed": false,
              "graph_traversal": undefined,
              "layout_name": "preset",
              "search_algorithm": undefined,
            },
            "specs": undefined,
            "tracks": [],
          },
          "da": null,
          "data": {},
          "devtoolsRawSetupState": {
            "cytoscape": RefImpl {
              "__v_isRef": true,
              "__v_isShallow": false,
              "_rawValue": null,
              "_value": null,
              "dep": undefined,
            },
            "dataset": RefImpl {
              "__v_isRef": true,
              "__v_isShallow": false,
              "_rawValue": undefined,
              "_value": undefined,
              "dep": undefined,
            },
            "getDataset": [Function],
            "props": {
              "datasetId": "MY_DATASET_ID",
              "datasetUrl": "network.json",
              "root": "/",
              "settings": {
                "color_picker_edges": "#A5A5A5",
                "color_picker_highlighted": "#C00000",
                "color_picker_nodes": "#548DB8",
                "curve_style": "haystack",
                "directed": false,
                "graph_traversal": undefined,
                "layout_name": "preset",
                "search_algorithm": undefined,
              },
              "specs": undefined,
              "tracks": [],
            },
            "render": [Function],
            "viewport": RefImpl {
              "__v_isRef": true,
              "__v_isShallow": false,
              "_rawValue": <div
                class="h-screen overflow-hidden"
                credentials="include"
              />,
              "_value": <div
                class="h-screen overflow-hidden"
                credentials="include"
              />,
              "dep": undefined,
            },
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "_depsLength": 0,
            "_dirtyLevel": 0,
            "_runnings": 0,
            "_shouldSchedule": false,
            "_trackId": 1,
            "active": true,
            "allowRecurse": true,
            "deps": [],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "scheduler": [Function],
            "trigger": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": {},
          "exposed": {},
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": [
            [Function],
          ],
          "next": null,
          "parent": [Circular],
          "props": {
            "datasetId": "MY_DATASET_ID",
            "datasetUrl": "network.json",
            "root": "/",
            "settings": {
              "color_picker_edges": "#A5A5A5",
              "color_picker_highlighted": "#C00000",
              "color_picker_nodes": "#548DB8",
              "curve_style": "haystack",
              "directed": false,
              "graph_traversal": undefined,
              "layout_name": "preset",
              "search_algorithm": undefined,
            },
            "specs": undefined,
            "tracks": [],
          },
          "propsDefaults": {},
          "propsOptions": [
            {
              "datasetId": {
                "0": false,
                "1": true,
                "required": true,
                "type": [Function],
              },
              "datasetUrl": {
                "0": false,
                "1": true,
                "required": true,
                "type": [Function],
              },
              "root": {
                "0": false,
                "1": true,
                "required": true,
                "type": [Function],
              },
              "settings": {
                "0": false,
                "1": true,
                "required": true,
                "type": [Function],
              },
              "specs": {
                "0": false,
                "1": true,
                "required": false,
                "type": [Function],
              },
              "tracks": {
                "0": false,
                "1": true,
                "required": true,
                "type": [Function],
              },
            },
            [],
          ],
          "provides": {},
          "proxy": {
            "datasetId": "MY_DATASET_ID",
            "datasetUrl": "network.json",
            "root": "/",
            "settings": {
              "color_picker_edges": "#A5A5A5",
              "color_picker_highlighted": "#C00000",
              "color_picker_nodes": "#548DB8",
              "curve_style": "haystack",
              "directed": false,
              "graph_traversal": undefined,
              "layout_name": "preset",
              "search_algorithm": undefined,
            },
            "specs": undefined,
            "tracks": [],
          },
          "refs": {
            "viewport": <div
              class="h-screen overflow-hidden"
              credentials="include"
            />,
          },
          "render": [Function],
          "renderCache": [],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "_active": true,
            "cleanups": [],
            "detached": true,
            "effects": [
              ReactiveEffect {
                "_depsLength": 20,
                "_dirtyLevel": 0,
                "_runnings": 0,
                "_shouldSchedule": false,
                "_trackId": 1,
                "active": true,
                "deps": [
                  Map {
                    [Circular] => 1,
                  },
                  Map {
                    [Circular] => 1,
                  },
                  Map {
                    [Circular] => 1,
                  },
                  Map {
                    [Circular] => 1,
                  },
                  Map {
                    [Circular] => 1,
                  },
                  Map {
                    [Circular] => 1,
                  },
                  Map {
                    [Circular] => 1,
                  },
                  Map {
                    [Circular] => 1,
                  },
                  Map {
                    [Circular] => 1,
                  },
                  Map {
                    [Circular] => 1,
                  },
                  Map {
                    [Circular] => 1,
                  },
                  Map {
                    [Circular] => 1,
                  },
                  Map {
                    [Circular] => 1,
                  },
                  Map {
                    [Circular] => 1,
                  },
                  Map {
                    [Circular] => 1,
                  },
                  Map {
                    [Circular] => 1,
                  },
                  Map {
                    [Circular] => 1,
                  },
                  Map {
                    [Circular] => 1,
                  },
                  Map {
                    [Circular] => 1,
                  },
                  Map {
                    [Circular] => 1,
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "scheduler": [Function],
                "trigger": [Function],
              },
              ReactiveEffect {
                "_depsLength": 0,
                "_dirtyLevel": 0,
                "_runnings": 0,
                "_shouldSchedule": false,
                "_trackId": 1,
                "active": true,
                "allowRecurse": true,
                "deps": [],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "scheduler": [Function],
                "trigger": [Function],
              },
            ],
            "parent": undefined,
          },
          "setupContext": {
            "attrs": {
              "credentials": "include",
            },
            "emit": [Function],
            "expose": [Function],
            "slots": {},
          },
          "setupState": {
            "cytoscape": null,
            "dataset": undefined,
            "getDataset": [Function],
            "props": {
              "datasetId": "MY_DATASET_ID",
              "datasetUrl": "network.json",
              "root": "/",
              "settings": {
                "color_picker_edges": "#A5A5A5",
                "color_picker_highlighted": "#C00000",
                "color_picker_nodes": "#548DB8",
                "curve_style": "haystack",
                "directed": false,
                "graph_traversal": undefined,
                "layout_name": "preset",
                "search_algorithm": undefined,
              },
              "specs": undefined,
              "tracks": [],
            },
            "render": [Function],
            "viewport": <div
              class="h-screen overflow-hidden"
              credentials="include"
            />,
          },
          "slots": {},
          "sp": null,
          "subTree": {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "ce": undefined,
            "children": null,
            "component": null,
            "ctx": [Circular],
            "dirs": null,
            "dynamicChildren": [],
            "dynamicProps": null,
            "el": <div
              class="h-screen overflow-hidden"
              credentials="include"
            />,
            "key": null,
            "patchFlag": 528,
            "props": {
              "class": "h-screen overflow-hidden",
              "credentials": "include",
              "ref": "viewport",
            },
            "ref": {
              "f": false,
              "i": [Circular],
              "k": undefined,
              "r": "viewport",
            },
            "scopeId": null,
            "shapeFlag": 1,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "targetStart": null,
            "transition": null,
            "type": "div",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": {
            "__file": "/Users/guerler/galaxy-visualizations/packages/cytoscape/src/Plugin.vue",
            "__name": "Plugin",
            "components": {},
            "props": {
              "datasetId": {
                "required": true,
                "type": [Function],
              },
              "datasetUrl": {
                "required": true,
                "type": [Function],
              },
              "root": {
                "required": true,
                "type": [Function],
              },
              "settings": {
                "required": true,
                "type": [Function],
              },
              "specs": {
                "required": false,
                "type": [Function],
              },
              "tracks": {
                "required": true,
                "type": [Function],
              },
            },
            "render": [Function],
            "setup": [Function],
          },
          "u": null,
          "uid": 1,
          "um": [
            [Function],
          ],
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "ctx": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="h-screen overflow-hidden"
          credentials="include"
        />,
        "key": null,
        "patchFlag": 0,
        "props": {
          "credentials": "include",
          "datasetId": "MY_DATASET_ID",
          "datasetUrl": "network.json",
          "ref": "VTU_COMPONENT",
          "root": "/",
          "settings": {
            "color_picker_edges": "#A5A5A5",
            "color_picker_highlighted": "#C00000",
            "color_picker_nodes": "#548DB8",
            "curve_style": "haystack",
            "directed": false,
            "graph_traversal": undefined,
            "layout_name": "preset",
            "search_algorithm": undefined,
          },
          "tracks": [],
        },
        "ref": {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "targetStart": null,
        "transition": null,
        "type": {
          "__file": "/Users/guerler/galaxy-visualizations/packages/cytoscape/src/Plugin.vue",
          "__name": "Plugin",
          "components": {},
          "props": {
            "datasetId": {
              "required": true,
              "type": [Function],
            },
            "datasetUrl": {
              "required": true,
              "type": [Function],
            },
            "root": {
              "required": true,
              "type": [Function],
            },
            "settings": {
              "required": true,
              "type": [Function],
            },
            "specs": {
              "required": false,
              "type": [Function],
            },
            "tracks": {
              "required": true,
              "type": [Function],
            },
          },
          "render": [Function],
          "setup": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": {
        "name": "VTU_ROOT",
        "render": [Function],
        "setup": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": {
          "app": [Circular],
          "components": {
            "transition": {
              "name": "transition",
            },
            "transition-group": {
              "name": "transition-group",
            },
          },
          "config": {
            "compilerOptions": {},
            "errorHandler": undefined,
            "globalProperties": {},
            "isNativeTag": [Function],
            "optionMergeStrategies": {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": {},
          "emitsCache": WeakMap {},
          "mixins": [
            {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "ctx": null,
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="h-screen overflow-hidden"
          credentials="include"
        />,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "targetStart": null,
        "transition": null,
        "type": {
          "name": "VTU_ROOT",
          "render": [Function],
          "setup": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": {
      "compilerOptions": {},
      "errorHandler": undefined,
      "globalProperties": {},
      "isNativeTag": [Function],
      "optionMergeStrategies": {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "runWithContext": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.4.38",
  },
  "__setProps": [Function],
  "cleanUpCallbacks": [
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
  ],
  "componentVM": {},
  "isDisabled": [Function],
  "rootVM": {
    "VTU_COMPONENT": {},
  },
  "wrapperElement": <div
    class="h-screen overflow-hidden"
    credentials="include"
  />,
}
`;
